packages/fireblocks-signing-manager/src/lib/fireblocks-signing-manager.ts

```diff
@@ -0,0 +1,175 @@
+    };
+  }
+
+  /**
+   * @hidden
+   *
+   * Get Fireblocks key info by its SS58 encoded public key
+   *
+   * @throws if there is no key has been derived with that address
+   */
+  private async getFireblocksKeyInfo(address: string): Promise<PublicKeyResonse> {
+    const pubKey = u8aToHex(decodeAddress(address));
+
+    const foundKey = this.fireblocks.lookupKey(pubKey);
+
+    if (!foundKey) {
+      throw new KeyNotFound('The signer cannot sign transactions on behalf of the calling Account');
+    }
+
+    return foundKey;
+  }
+}
+
+export class FireblocksSigningManager implements SigningManager {
+  public fireblocksClient: Fireblocks;
+  private externalSigner: FireblocksSigner;
+  private _ss58Format?: number;
+
+  /**
+   * @hidden
+   */
+  private constructor(args: Omit<CreateParams, 'derivationPaths'>) {
+    const { url, apiKey, secretPath } = args;
+
+    this.fireblocksClient = new Fireblocks({ url, apiKey, secretPath });
+    this.externalSigner = new FireblocksSigner(this.fireblocksClient, new TypeRegistry());
+  }
+
+  public static async create(args: CreateParams): Promise<FireblocksSigningManager> {
+    const signingManager = new FireblocksSigningManager(args);
+
+    if (args.derivationPaths) {
+      if (args.derivationPaths.length > maxInitialDerivedAccounts) {
+        throw new ConfigError(
+          `Number of initial derivation paths cannot exceed ${maxInitialDerivedAccounts}. Use deriveAccount after creation to load more accounts instead`
+        );
+      }
+
+      const derivePromises = args.derivationPaths.map(path =>
+        signingManager.fireblocksClient.deriveKey(path)
+      );
+      await Promise.all(derivePromises);
+    } else {
+      await signingManager.fireblocksClient.deriveKey(defaultKeyPath);
+    }
+
+    return signingManager;
+  }
+
+  /**
+   * Set the SS58 format in which addresses will be encoded
+   */
+  public setSs58Format(ss58Format: number): void {
+    this._ss58Format = ss58Format;
+  }
+
+  public get ss58Format(): number {
+    if (!this._ss58Format) {
+      throw new ConfigError(
+        'FireblocksSigningManager ss58Format was not set. The Polymesh SDK should set the format upon its initialization'
+      );
+    }
+
+    return this._ss58Format;
+  }
+
+  /**
+   * Return the addresses of all derived keys in Fireblocks
+   */
+  public async getAccounts(): Promise<string[]> {
+    const allKeys = this.fireblocksClient.fetchDerivedKeys();
+
+    const ss58Format = this.ss58Format;
+
+    return allKeys.map(({ publicKey }) => encodeAddress(`0x${publicKey}`, ss58Format));
+  }
+
+  /**
+   * Derive an Account to be used for signing
+   *
+   * @note this method must be called with the derivation path for non default addresses before they can be used to sign
+   */
+  public async deriveAccount(path: DerivationPath): Promise<KeyInfo> {
+    const key = await this.fireblocksClient.deriveKey(path);
+
+    const address = encodeAddress(`0x${key.publicKey}`, this.ss58Format);
```

Comment:
Nit-> you can use the `hexAddPrefix` from polkadot util to add `0x`. That way if the public key itself adds in 0x somewhere then you don't have to change a thing